OpenRA.Mods.Common/Traits/Render/LeavesTrails.cs

```diff
@@ -109,11 +106,15 @@ public void Tick(Actor self)
 				var spawnCell = Info.SpawnAtLastPosition ? self.World.Map.CellContaining(cachedPosition) : self.World.Map.CellContaining(self.CenterPosition);
 				var type = self.World.Map.GetTerrainInfo(spawnCell).Type;
 
-				var offsetRotation = Info.Offset.Rotate(body.QuantizeOrientation(self, self.Orientation));
-				var spawnOffset = body.LocalToWorld(Info.AlternateOffset && evenNumber ? -offsetRotation : offsetRotation);
+				if (offset < Info.Offsets.Length)
```

Comment:
If `offset == Info.Offsets.Length - 1` it will pass this check and be incremented to `Info.Offsets.Length`, which then gives your OutOfBoundsException.

You can simplify this if/else block to:

```csharp
if (++offset >= Info.Offsets.Length)
    offset = 0;
```